pkg/nodeconfig/nodeconfig.go

```diff
@@ -44,8 +44,8 @@ const (
 // nodeConfig holds the information to make the given VM a kubernetes node. As of now, it holds the information
 // related to kubeclient and the windowsVM.
 type nodeConfig struct {
-	// k8sclientset holds the information related to kubernetes clientset
-	k8sclientset *kubernetes.Clientset
```

Comment:
Looks like we will need the kubernetes clientset for using the [drain helper](https://pkg.go.dev/k8s.io/kubectl/pkg/drain#Helper) which will be used to cordon the node. Given that I suggest not removing this.